### PR TITLE
#161290919 Users Reading Stats

### PR DESCRIPTION
**Pull request by ajimae**
#### What does this PR do?
 Enable users to see their reading statistics

#### Description of Task to be completed?

- Write test
- Add a migration script to create a ReadingStats table
- Add a model 
- Write functions to post reading statistics to the ReadingStats table
- Write tests to validate functionality
- Create a route to handle route users get reading statistics requests
- Write functions to get users reading statistics
- Validate functionality in Postman

#### How should this be manually tested?

- Clone the repo to local machine
- Run `npm install` to install dependencies and devDependencies
- Run Postma, using the `POST` verb visit the address `localhost:3000/api/v1/users/register` to register an account
- Visit `localhost:3000/api/v1/articles` provide the necessary request body to post an article using `POST`
- Get an article by visiting the `URL` `localhost:3000/api/v1/articles` using `GET`

#### Any background context you want to provide?

- Modified the `Procfile` to drop all database tables on before each build.
- Refactored the `verificationWorker.js` file

#### What are the relevant pivotal tracker stories?

[#161290919](https://www.pivotaltracker.com/story/show/161290919)

#### Screenshots

screen shot 2018-11-28 at 2 26 45 pm

### CODE REVIEW

**Review by hope-uwa — CHANGES_REQUESTED** [review]
@ajimae I noticed this PR does more than just implementing the Users Reading stat. Please state the other things it does in the PR description, example the bug fix for verification middleware. And also noticed some spelling errors in ReadingStatsWorker on ln13 and ln39.


Thanks so much for your observation, feedback was well received @uwaelpis, also, please you are not specific enough. Where (on what file) did you see the spelling errors? and besides the changes on the `verificationWorker.js` wasn't a bug fix it was a minor refactor. Also as a feedback, I suggest you pay more attention to details and understand what is going on before you comment. Thanks once again. Please direct me to where the spelling errors are, so I can make the necessary changes.

File: server/src/workers/ReadingStatsWorker.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+const {
+  Article,
+  ReadingStats,
+} = models;
+
+/**
+ * @class { @ }
+ * @description { Creation of Users reading statistics }
+ */
+class ReadingSatsWorker {
```
Comment (hope-uwa) [inline server/src/workers/ReadingStatsWorker.js]: I guess you meant to spell ReadingStatsWorker

File: server/src/workers/ReadingStatsWorker.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    return getArticle;
+  }
+
+  /**
+   * @description { A function that the details of an article from the database }
+   * @param { object } userId
+   * @param { object } res
+   * @returns { object } getArticle
+   */
+  static async getUserRedingStats(userId, res) {
```
Comment (hope-uwa) [inline server/src/workers/ReadingStatsWorker.js]: and here too getUserReadingStats

**Comment by hope-uwa** [comment]
@ajimae Thanks for the feedback, You should have captured the minor refactoring done in the PR description, because I see a lot of file changes that are not stated in the description. 


Done.
But I think it should go to the ### Background context section, so I entered it there.
Thanks for the feedback

File: server/src/workers/ReadingStatsWorker.js (hunk 1 above)
Comment (ajimae) [reply server/src/workers/ReadingStatsWorker.js]: Done.
Thanks for pointing it out.

File: server/src/workers/ReadingStatsWorker.js (hunk 2 above)
Comment (ajimae) [reply server/src/workers/ReadingStatsWorker.js]: Done.
Thanks for pointing it out also.

**Review by hope-uwa — APPROVED** [review]

**Comment by hope-uwa** [comment]
Great job!

**Review by ovieokeh — APPROVED** [review]

**Pull request merged**